### ISSUE TITLE
tf.nn.fractional_max_pool aborts when pooling_ratio is negative

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#42278]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

`tf.nn.fractional_max_pool` aborts when `pooling_ratio` is negative in tf2.1. It throws segfault in nightly version.

**Describe the expected behavior**
Expect no crashes or aborts. 
**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
~~~python
import tensorflow as tf
tf.nn.fractional_max_pool(value=tf.ones((1,1,1,1)), seed=1, pooling_ratio=-1)
~~~
**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
Error message in TF2.1:
~~~python
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
~~~
Error message in TF nightly:
~~~
Segmentation fault (core dumped)
~~~

**Comment by ravikyram** [comment]
@DNXie 

This is intended behavior.`Pooling ratio for each dimension of value, currently only supports row and col dimension and should be >= 1.0.`.
Please, refer the document [here](https://www.tensorflow.org/api_docs/python/tf/nn/fractional_max_pool#args).I i take value pooling ration =1 i am not seeing any issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/a37924989af91984416d6043aa64f0d2/untitled249.ipynb).Thanks!

**Comment by aavishkarmishra** [comment]
Hi, I would like to work on this can anyone help me with how it should behave with `pooling ratio<1` as I am new in this.

**Comment by DNXie** [comment]
@ravikyram @aavishkarmishra  It should throw an exception instead of crash.

**Comment by aavishkarmishra** [comment]
So, mainly test pooling ratio to be>=1 and throw error if it is not.

**Comment by DNXie** [comment]
@aavishkarmishra I am not an expert, but some exception message thrown in python level would be good.

**Comment by Saduf2019** [comment]
I ran the code on nightly [2.4.0-dev20200916] and the issue still persist. please find the [gist here](https://colab.research.google.com/gist/Saduf2019/ecc94688daba6d6f4f7a73ce8923ae0d/untitled415.ipynb).

**Comment by aavishkarmishra** [comment]
@ravikyram This issue is resolved after the merge of pull request please close this issue.
